### ISSUE TITLE
[YSQL] Further push down = operator for range key

### ISSUE THREAD

**Issue by jaki** [yugabyte/yugabyte-db#3309]
Although the `=` (equals) operator is pushed down such that PostgreSQL does not need to filter through all rows of a table, it could be pushed down further.  Observe this difference between hash and range keys.

```sql
CREATE TABLE h (i int PRIMARY KEY);
INSERT INTO h VALUES (1), (2), (3);
SELECT * FROM h WHERE i = 2;

CREATE TABLE r (i int, PRIMARY KEY (i ASC));
INSERT INTO r VALUES (1), (2), (3);
SELECT * FROM r WHERE i = 2;
```

On `SELECT` to `h`, the bounds given in the `doc_spec` parameter of `DocRowwiseIterator::DoInit` are `DocKey(0xc0c4, [2], []), DocKey(0xc0c4, [2], [+Inf])`.  On `SELECT` to `r`, the bounds are `DocKey([], []), DocKey([], [+Inf])`.  For the `SELECT` to `r`, the bounds should rather be `DocKey([], [2]), DocKey([], [2, +Inf])`.

`PgsqlReadOperation::Execute` filters out rows using `request_.where_expr()`.  What I am saying is that, before getting to that step, the iterator created should already be bound like I proposed above.

**Comment by ndeodhar** [comment]
This is important for colocated tables which will use range PK for tables.

**Comment by ndeodhar** [comment]
Fixed as a result of #4033
